TechnologyAssessmentApp/TechnologyAssessmentApp/Controllers/DetailViewController.swift

```diff
@@ -14,34 +14,23 @@ class DetailViewController: UIViewController {
     // MARK: - IBOutlets
     //
     @IBOutlet weak var detailLabel: UILabel!
-    var article: Article? {
+    var article: Result? {
         didSet {
             refreshUI()
         }
     }
     func refreshUI() {
         loadViewIfNeeded()
-        if let art = self.article, let abstractString = art.abstract {
-             self.detailLabel.text = abstractString
+        if let art = article {
+             detailLabel.text = art.abstract
         }
     }
-
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destinationViewController.
-        // Pass the selected object to the new view controller.
-    }
-    */
-
 }
 
-extension DetailViewController: ArticleSelectionDelegate {
-    func articleSelected(_ articleDetail: Article?) {
+extension DetailViewController: MasterTableViewSelectionDelegate {
+    func articleSelected(_ articleDetail: Result?) {
```

Comment:
is this delegate supposed to be called with `articleDetail` which is `nil` ? I don't think so, so it should not be optional. Call the delegate only if article exist.